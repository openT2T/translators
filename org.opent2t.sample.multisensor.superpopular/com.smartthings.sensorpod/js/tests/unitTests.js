var translatorPath = require('path').join(__dirname, '..');
var runMultisensorTests = require('opent2t-device-multisensor/multisensorTests');
var deviceData = require('./devicedata');
var testData = require('./testdata');
var MockHub = require('opent2t-device-smartthingshub/mockSmartthingsHub');
var mockHub = new MockHub(deviceData);

var settings = {
    translatorPath,
    getDeviceInfo: mockHub.getDeviceInfo.bind(mockHub),
    setTestData: mockHub.setTestData.bind(mockHub),
    expectedExceptions : testData.expected_exceptions,
};

// Run standard lamp unit tests
runMultisensorTests(settings);