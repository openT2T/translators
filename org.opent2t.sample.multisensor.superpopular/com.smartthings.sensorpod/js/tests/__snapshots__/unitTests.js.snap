exports[`GetPlatform 1`] = `"{"opent2t":{"schema":"org.opent2t.sample.multisensor.superpopular","translator":"opent2t-translator-com-smartthings-sensorpod","controlId":"c07d4da8-18f2-4e93-9259-e96fdc8fb636","endpointUri":"https://sample.endpoint.com/123"},"pi":"c07d4da8-18f2-4e93-9259-e96fdc8fb636","mnmn":"Mock Manufacturer","mnmo":"Mock Model","n":"Mock Name","rt":["org.opent2t.sample.multisensor.superpopular"],"entities":[{"n":"Mock Name","rt":["opent2t.d.sensor.contact"],"di":"67D227BD-376A-4DA8-BAB5-90E95B977EE0","icv":"core.1.1.0","dmv":"res.1.1.0","resources":[{"href":"/contact","rt":["oic.r.sensor.contact"],"if":["oic.if.s","oic.if.baseline"]},{"href":"/lastchanged","rt":["opent2t.r.timestamp"],"if":["oic.if.s","oic.if.baseline"]}]},{"n":"Mock Name","rt":["opent2t.d.sensor.locked"],"di":"B9B3E572-0FB7-4F9D-A8B1-BF90DB120FBB","icv":"core.1.1.0","dmv":"res.1.1.0","resources":[{"href":"/locked","rt":["oic.r.sensor"],"if":["oic.if.s","oic.if.baseline"]},{"href":"/lastchanged","rt":["opent2t.r.timestamp"],"if":["oic.if.s","oic.if.baseline"]}]},{"n":"Mock Name","rt":["opent2t.d.sensor.loudnesschange"],"di":"BC47CB27-C234-46D7-B03F-8CDC1D52F5B3","icv":"core.1.1.0","dmv":"res.1.1.0","resources":[{"href":"/loudnesschange","rt":["oic.r.sensor"],"if":["oic.if.s","oic.if.baseline"]},{"href":"/lastchanged","rt":["opent2t.r.timestamp"],"if":["oic.if.s","oic.if.baseline"]}]},{"n":"Mock Name","rt":["opent2t.d.sensor.motion"],"di":"63E5A41E-7283-4BA6-A3A8-21AE3C18C17F","icv":"core.1.1.0","dmv":"res.1.1.0","resources":[{"href":"/motion","rt":["oic.r.sensor.motion"],"if":["oic.if.s","oic.if.baseline"]},{"href":"/lastchanged","rt":["opent2t.r.timestamp"],"if":["oic.if.s","oic.if.baseline"]}]},{"n":"Mock Name","rt":["opent2t.d.sensor.presence"],"di":"F0FDF054-C8FF-44CD-AE1D-BD129AB4FF99","icv":"core.1.1.0","dmv":"res.1.1.0","resources":[{"href":"/presence","rt":["oic.r.sensor.presence"],"if":["oic.if.s","oic.if.baseline"]},{"href":"/lastchanged","rt":["opent2t.r.timestamp"],"if":["oic.if.s","oic.if.baseline"]}]},{"n":"Mock Name","rt":["opent2t.d.sensor.temperature"],"di":"E7FA6B8B-4B8B-4172-840A-00414BA5055E","icv":"core.1.1.0","dmv":"res.1.1.0","resources":[{"href":"/temperature","rt":["oic.r.temperature"],"if":["oic.if.s","oic.if.baseline"]},{"href":"/lastchanged","rt":["opent2t.r.timestamp"],"if":["oic.if.s","oic.if.baseline"]}]},{"n":"Mock Name","rt":["opent2t.d.sensor.vibrationchange"],"di":"D38E580A-C4E2-45C5-A316-5A685553B868","icv":"core.1.1.0","dmv":"res.1.1.0","resources":[{"href":"/vibrationchange","rt":["oic.r.sensor"],"if":["oic.if.s","oic.if.baseline"]},{"href":"/lastchanged","rt":["opent2t.r.timestamp"],"if":["oic.if.s","oic.if.baseline"]}]},{"n":"Mock Name","rt":["opent2t.d.sensor.water"],"di":"6726E344-8471-4A99-90D4-403EFD961936","icv":"core.1.1.0","dmv":"res.1.1.0","resources":[{"href":"/water","rt":["oic.r.sensor.water"],"if":["oic.if.s","oic.if.baseline"]},{"href":"/lastchanged","rt":["opent2t.r.timestamp"],"if":["oic.if.s","oic.if.baseline"]}]},{"n":"Mock Name","rt":["opent2t.d.battery"],"di":"06C47089-1049-46CF-BB64-74F7E7C4F501","icv":"core.1.1.0","dmv":"res.1.1.0","resources":[{"href":"/battery","rt":["oic.r.energy.battery"],"if":["oic.if.s","oic.if.baseline"]}]}]}"`;

exports[`GetPlatformExpanded 1`] = `"{"opent2t":{"schema":"org.opent2t.sample.multisensor.superpopular","translator":"opent2t-translator-com-smartthings-sensorpod","controlId":"c07d4da8-18f2-4e93-9259-e96fdc8fb636","endpointUri":"https://sample.endpoint.com/123"},"pi":"c07d4da8-18f2-4e93-9259-e96fdc8fb636","mnmn":"Mock Manufacturer","mnmo":"Mock Model","n":"Mock Name","rt":["org.opent2t.sample.multisensor.superpopular"],"entities":[{"n":"Mock Name","rt":["opent2t.d.sensor.contact"],"di":"67D227BD-376A-4DA8-BAB5-90E95B977EE0","icv":"core.1.1.0","dmv":"res.1.1.0","resources":[{"href":"/contact","rt":["oic.r.sensor.contact"],"if":["oic.if.s","oic.if.baseline"],"id":"contact","value":false},{"href":"/lastchanged","rt":["opent2t.r.timestamp"],"if":["oic.if.s","oic.if.baseline"],"id":"lastchanged","timestamp":"2017-03-15T05:31:12.305Z"}]},{"n":"Mock Name","rt":["opent2t.d.sensor.locked"],"di":"B9B3E572-0FB7-4F9D-A8B1-BF90DB120FBB","icv":"core.1.1.0","dmv":"res.1.1.0","resources":[{"href":"/locked","rt":["oic.r.sensor"],"if":["oic.if.s","oic.if.baseline"],"id":"locked","value":true},{"href":"/lastchanged","rt":["opent2t.r.timestamp"],"if":["oic.if.s","oic.if.baseline"],"id":"lastchanged","timestamp":"2017-03-14T05:32:45.325Z"}]},{"n":"Mock Name","rt":["opent2t.d.sensor.loudnesschange"],"di":"BC47CB27-C234-46D7-B03F-8CDC1D52F5B3","icv":"core.1.1.0","dmv":"res.1.1.0","resources":[{"href":"/loudnesschange","rt":["oic.r.sensor"],"if":["oic.if.s","oic.if.baseline"],"id":"loudnesschange","value":true},{"href":"/lastchanged","rt":["opent2t.r.timestamp"],"if":["oic.if.s","oic.if.baseline"],"id":"lastchanged","timestamp":"2016-10-25T09:04:49.858Z"}]},{"n":"Mock Name","rt":["opent2t.d.sensor.motion"],"di":"63E5A41E-7283-4BA6-A3A8-21AE3C18C17F","icv":"core.1.1.0","dmv":"res.1.1.0","resources":[{"href":"/motion","rt":["oic.r.sensor.motion"],"if":["oic.if.s","oic.if.baseline"],"id":"motion","value":false},{"href":"/lastchanged","rt":["opent2t.r.timestamp"],"if":["oic.if.s","oic.if.baseline"],"id":"lastchanged","timestamp":"2017-03-15T03:12:30.415Z"}]},{"n":"Mock Name","rt":["opent2t.d.sensor.presence"],"di":"F0FDF054-C8FF-44CD-AE1D-BD129AB4FF99","icv":"core.1.1.0","dmv":"res.1.1.0","resources":[{"href":"/presence","rt":["oic.r.sensor.presence"],"if":["oic.if.s","oic.if.baseline"],"id":"presence","value":true},{"href":"/lastchanged","rt":["opent2t.r.timestamp"],"if":["oic.if.s","oic.if.baseline"],"id":"lastchanged","timestamp":"2017-03-14T02:20:31.465Z"}]},{"n":"Mock Name","rt":["opent2t.d.sensor.temperature"],"di":"E7FA6B8B-4B8B-4172-840A-00414BA5055E","icv":"core.1.1.0","dmv":"res.1.1.0","resources":[{"href":"/temperature","rt":["oic.r.temperature"],"if":["oic.if.s","oic.if.baseline"],"id":"temperature","temperature":"73","units":"f"},{"href":"/lastchanged","rt":["opent2t.r.timestamp"],"if":["oic.if.s","oic.if.baseline"],"id":"lastchanged","timestamp":"2017-03-15T03:54:59.836Z"}]},{"n":"Mock Name","rt":["opent2t.d.sensor.vibrationchange"],"di":"D38E580A-C4E2-45C5-A316-5A685553B868","icv":"core.1.1.0","dmv":"res.1.1.0","resources":[{"href":"/vibrationchange","rt":["oic.r.sensor"],"if":["oic.if.s","oic.if.baseline"],"id":"vibrationchange","value":false},{"href":"/lastchanged","rt":["opent2t.r.timestamp"],"if":["oic.if.s","oic.if.baseline"],"id":"lastchanged","timestamp":"2017-03-15T05:47:01.990Z"}]},{"n":"Mock Name","rt":["opent2t.d.sensor.water"],"di":"6726E344-8471-4A99-90D4-403EFD961936","icv":"core.1.1.0","dmv":"res.1.1.0","resources":[{"href":"/water","rt":["oic.r.sensor.water"],"if":["oic.if.s","oic.if.baseline"],"id":"water","value":false},{"href":"/lastchanged","rt":["opent2t.r.timestamp"],"if":["oic.if.s","oic.if.baseline"],"id":"lastchanged","timestamp":"2016-10-25T10:06:36.756Z"}]},{"n":"Mock Name","rt":["opent2t.d.battery"],"di":"06C47089-1049-46CF-BB64-74F7E7C4F501","icv":"core.1.1.0","dmv":"res.1.1.0","resources":[{"href":"/battery","rt":["oic.r.energy.battery"],"if":["oic.if.s","oic.if.baseline"],"id":"battery","charge":"67"}]}]}"`;
